src/lxc/console.c

```diff
@@ -121,27 +121,29 @@ struct lxc_tty_state *lxc_console_sigwinch_init(int srcfd, int dstfd)
 	ts->masterfd = dstfd;
 	ts->sigfd = -1;
 
-	/* add tty to list to be scanned at SIGWINCH time */
-	lxc_list_add_elem(&ts->node, ts);
-	lxc_list_add_tail(&lxc_ttys, &ts->node);
+	if (isatty(srcfd)) {
+		/* add tty to list to be scanned at SIGWINCH time */
+		lxc_list_add_elem(&ts->node, ts);
+		lxc_list_add_tail(&lxc_ttys, &ts->node);
+
+		sigemptyset(&mask);
+		sigaddset(&mask, SIGWINCH);
+		if (sigprocmask(SIG_BLOCK, &mask, &ts->oldmask)) {
+			SYSERROR("failed to block SIGWINCH.");
+			ts->sigfd = -1;
+			return ts;
+		}
 
-	sigemptyset(&mask);
-	sigaddset(&mask, SIGWINCH);
-	if (sigprocmask(SIG_BLOCK, &mask, &ts->oldmask)) {
-		SYSERROR("failed to block SIGWINCH.");
-		ts->sigfd = -1;
-		return ts;
-	}
+		ts->sigfd = signalfd(-1, &mask, 0);
+		if (ts->sigfd < 0) {
+			SYSERROR("failed to get signalfd.");
+			sigprocmask(SIG_SETMASK, &ts->oldmask, NULL);
+			ts->sigfd = -1;
+			return ts;
+		}
 
-	ts->sigfd = signalfd(-1, &mask, 0);
-	if (ts->sigfd < 0) {
-		SYSERROR("failed to get signalfd.");
-		sigprocmask(SIG_SETMASK, &ts->oldmask, NULL);
-		ts->sigfd = -1;
-		return ts;
+		DEBUG("%d got SIGWINCH fd %d", getpid(), ts->sigfd);
```

Comment:
While you're at it could you change this to:

    DEBUG("process %d created signal fd %d to handle SIGWINCH events", getpid(), ts->sigfd)